### ISSUE TITLE
False positive for ``super-init-not-called``

### ISSUE THREAD

**Issue by DanielNoord** [PyCQA/pylint#4941]
### Bug description

```python
class A:
    def __init__(self):
        pass


class B(A):
    pass


class D(B, A):
    def __init__(self):
        A.__init__(self)
```

This was found while working on #4940, after fixing this we might want to check if there are any messages that are unnecessarily disabled in our own codebase.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
pylint test.py
************* Module test
test.py:11:4: W0231: __init__ method from base class 'B' is not called (super-init-not-called)

------------------------------------------------------------------
Your code has been rated at 8.75/10 (previous run: 8.75/10, +0.00)
```

### Expected behavior

As class `B` does not have a `__init__` it can't be called. The only `__init__` included in class `B` is the `__init__` of class `A` which is being called in class `C`.

### Pylint version

```shell
pylint 2.10.3-dev0
astroid 2.7.3
Python 3.9.6 (default, Aug  2 2021, 08:23:35) 
[Clang 12.0.5 (clang-1205.0.22.11)]
```

### OS / Environment

macOS

**Comment by Goldziher** [comment]
Another example:

```python
class DataclassProtocol(Protocol):
    def __init__(self, **kwargs):  # pylint: disable=super-init-not-called
        ...

    __dataclass_fields__: Dict
    __dataclass_params__: Dict
    __post_init__: Optional[Callable]

```

**Comment by DanielNoord** [comment]
@Goldziher You're issue was being tracked in #4790 for which I just created a fix #5697. Should be available in `2.13` 😄
